chatsky/processing/standard.py

```diff
@@ -39,3 +46,45 @@ async def call(self, ctx: Context) -> MessageInitTypes:
                 return await processing_object.modified_response(current_response, ctx)
 
         ctx.current_node.response = ModifiedResponse()
+
+
+class AddFallbackResponses(ModifyResponse, arbitrary_types_allowed=True):
+    """
+    ModifyResponse with pre-response processing to handle exceptions dynamically.
+    """
+
+    exception_response: Dict[Union[Type[Exception], Literal["Else"]], AnyResponse]
+    """
+    Dictionary mapping exception types to fallback responses.
+    """
+
+    @field_validator("exception_response")
+    @classmethod
+    def validate_not_empty(cls, exception_response: dict) -> dict:
+        """
+        Validate that the `exception_response` dictionary is not empty.
+
+        :param exception_response: Dictionary mapping exception types to fallback responses.
+        :raises ValueError: If the `exception_response` dictionary is empty.
+        :return: Not empty dictionary of exception_response.
+        """
+        if len(exception_response) == 0:
+            raise ValueError("Exceptions dict is empty")
+        return exception_response
+
+    async def modified_response(self, original_response: BaseResponse, ctx: Context) -> MessageInitTypes:
+        """
+        Catch response errors and process them based on `exception_response`.
+
+        :param original_response: The original response of the current node.
+        :param ctx: The current context.
+
+        :return: Message to replace original response with modified due to fallback response.
+        """
+        try:
+            return await original_response(ctx)
+        except Exception as e:
+            exception = self.exception_response.get(type(e), self.exception_response.get("Else"))
+            logger.exception(e)
+            ctx.framework_data.response_exception = str(e)
+            return await exception(ctx)
```

Comment:
I think it's better to do
```suggestion
        result = await original_response.wrapped_call(ctx)
        if isinstance(result, Exception):
            exception = self.exception_response.get(type(result), self.exception_response.get("Else"))
            ctx.framework_data.response_exception = str(result)
            return await exception(ctx)
        else:
            return result
```

in order to call wrapped call with its exception processing to account for the possibility of a different exception processing in response (e.g. if BaseScriptFunction is subclassed and `wrapped_call` is updated to have a different way to handle exceptions).